calabash-cucumber/lib/calabash-cucumber/launch/simulator_launcher.rb

```diff
@@ -251,13 +251,27 @@ def app_bundle_or_raise(path=nil, device_build_dir='iPhoneSimulator')
             puts('-'*37)
           end
         else
-          bo_dir = build_output_dir_for_project
-          sim_dirs = Dir.glob(File.join(bo_dir, '*-iphonesimulator', '*.app'))
+          search_dir = build_output_dir_for_project || DERIVED_DATA
+          sim_dirs = ''
+          apps = `find #{search_dir} -type d -name "*.app" -exec stat -f "%m %N" {} \\; | sort -rn | cut -d" " -f2`.split("\n")
```

Comment:
I think we need to be careful of UTF-8 chars here.

We can use ruby's Find or Dir.glob + File.mtime to perform this operation.